res/controllers/common-hid-packet-parser.js

```diff
@@ -1,1549 +1,1830 @@
+ * @property {boolean} ignored
+ * @property {controlCallback} auto_repeat
+ * @property {number} auto_repeat_interval
+ * @property {number} min
+ * @property {number} max
+ * @property {('bitvector'|'button'|'control'|'output')} type Must be either:
+ *              - 'bitvector'       If value is of type HIDBitVector
+ *              - 'button'          If value is a boolean
+ *              - 'control'         If value is a number
+ *              - 'output'
+ * @property {HIDBitVector|boolean|number} value
+ * @property {number} delta
+ * @property {number} mindelta
+ * @property {number} toggle
+ */
 
-/** HID Bit Vector Class
+/**
+ * @typedef bitObject
+ * @type {object}
+ * @property {HIDPacket} packet
+ * @property {string} id Group and control name separated by a dot
+ * @property {string} group
+ * @property {string} name
+ * @property {string} mapped_group
+ * @property {string} mapped_name
+ * @property {number} bitmask
+ * @property {number} bit_offset
+ * @property {controlCallback} callback
+ * @property {controlCallback} auto_repeat
+ * @property {number} auto_repeat_interval
+ * @property {('button'|'output')} type Must be either:
+ *              - 'button'
+ *              - 'output'
+ * @property {boolean} value
+ */
+
+/**
+ * HID Bit Vector Class
  *
  * Collection of bits in one parsed packet field. These objects are
  * created by HIDPacket addControl and addOutput and should not be
- * created manually. */
-var HIDBitVector = function() {
-    this.size = 0;
-    this.bits = {};
-};
-
-/** Return bit offset based on bitmask */
-HIDBitVector.prototype.getOffset = function(bitmask) {
-    for (var i = 0; i < 32; i++)
-        if ((1 & bitmask >> i) !== 0)
-            return i;
-    return 0;
-};
-
-/** Add a control bitmask to the HIDBitVector */
-HIDBitVector.prototype.addBitMask = function(group, name, bitmask) {
-    var bit = {};
-    bit.type = "button";
-    bit.packet = undefined;
-    bit.id = group + "." + name;
-    bit.group = group;
-    bit.name = name;
-    bit.mapped_group = undefined;
-    bit.mapped_name = undefined;
-    bit.bitmask = bitmask;
-    bit.bitmask = bitmask;
-    bit.bit_offset = this.getOffset(bitmask);
-    bit.callback = undefined;
-    bit.value = undefined;
-    bit.auto_repeat = undefined;
-    bit.auto_repeat_interval = undefined;
-    this.bits[bit.id] = bit;
-};
+ * created manually.
+ *
+ * @property {number} size
+ * @property {bitObject{}} bits
+ */
+class HIDBitVector {
+    constructor() {
+        this.size = 0;
+        this.bits = {};
+    }
+    /**
+     * Get the index of the least significant bit that is 1 in `bitmask`
+     *
+     * @param {number} bitmask A bitwise mask of up to 32 bit. All bits set to'1' in this mask are considered.
+     * @returns {number} Index of the least significant bit that is 1 in `bitmask`
+     */
+    getOffset(bitmask) {
+        for (let i = 0; i < 32; i++) {
+            if ((1 & bitmask >> i) !== 0) { return i; }
```

Comment:
This is hard to understand. Is it 
```suggestion
            if ((bitmask >> i) & 1) { 
                return i; 
            }
```